UC NetCode Fall 2024/Assets/App/Resource/Scripts/ProjectileObjs.cs

```diff
@@ -0,0 +1,42 @@
+using App.Resource.Scripts.Player;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace App.Resource.Scripts.obj
+{
+    public class ProjectileObjs : NetworkBehaviour
+    {
+
+        [SerializeField] float _speed = 40f;
+        [SerializeField] private float _damage = 10;
+        [SerializeField] private float destructTime = 5f;
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            GetComponent<Rigidbody>().velocity = this.transform.forward * _speed;
+            StartCoroutine(routine: AutoDestruct());
+
+        }
+
+        private void OnCollisionEnter(Collision other)
+        {
+            if (other.gameObject.tag.Equals("Player") && other.gameObject.GetComponent<NetworkObject>().OwnerClientId != this.OwnerClientId) ;
```

Comment:
there's an extra` ;` at the end of if `(other.gameObject.tag.Equals("Player") && other.gameObject.GetComponent<NetworkObject>().OwnerClientId != this.OwnerClientId) ;` you need to remove the ; in order to properly get the results of the if condition